tests/lit/single/pointer/referenceto_variable_uninitialized_address.st

```diff
@@ -0,0 +1,22 @@
+// RUN: (%COMPILE %s && %RUN) | %CHECK %s
+FUNCTION main : DINT
+    VAR
+        foo : REFERENCE TO DINT;
+        bar : REFERENCE TO DINT;
+        qux : DINT;
+    END_VAR
+
+    foo REF= bar;
+
+    // CHECK-NOT: ^5$
+    printf('%d$N', foo); // We expect some random value here because foo isn't pointing to any variable yet
```

Comment:
we are auto-dereferencing an uninitialized pointer here - this is undefined behaviour. Should we nitialize these pointers with `null` like we are doing for `VAR_INPUT {ref}`? Ideally we would check for use before assignment but that feature feels a bit far off still.

Reply:
Discussed offline, generally in favor of NULL pointers here but I'd probably also address this in another PR. To have a second opinion here though cc @ghaith

Reply:
I think null makes sense here so that we don't have random values. I'm in favour of not having such tests tbh. We can't test undefined behaviour (by design). Also what happens if the random value is 5? I am more in favour of having a test that verifies the pointer value of foo is null. Not sure how to do this with the auto deref rules.. maybe it's enough to have an IR check that verifies the initial values of REFERENCE TO and Pointer fields in general.

Reply:
Sounds good, I've created an issue and deleted the test here. I'll address it in another PR since this one bloats up way too much if you'd ask me.